src/nvim/lua/executor.c

```diff
@@ -1465,3 +1465,40 @@ void nlua_free_typval_dict(dict_T *const d)
     d->lua_table_ref = LUA_NOREF;
   }
 }
+
+void nlua_execute_log_keystroke(int c)
+{
+  char_u buf[NUMBUFLEN];
+  buf[special_to_buf(c, mod_mask, false, buf)] = NUL;
```

Comment:
Instead of doing this, you could do 
```suggestion
  int buf_len = special_to_buf(c, mod_mask, false, buf);
```
and then further down use `lua_pushlstring(lstate, buf, buf_len)` which precalculates the buffer length.